src/EditorFeatures/Core/Implementation/Classification/ClassificationUtilities.cs

```diff
@@ -17,46 +14,19 @@ namespace Microsoft.CodeAnalysis.Editor.Implementation.Classification
 {
     internal static class ClassificationUtilities
     {
-        private static readonly ConcurrentQueue<List<ClassifiedSpan>> s_spanCache = new();
-
-        public static List<ClassifiedSpan> GetOrCreateClassifiedSpanList()
-        {
-            return s_spanCache.TryDequeue(out var result)
-                ? result
-                : new List<ClassifiedSpan>();
-        }
-
-        public static void ReturnClassifiedSpanList(List<ClassifiedSpan> list)
-        {
-            if (list == null)
-            {
-                return;
-            }
-
-            list.Clear();
-            s_spanCache.Enqueue(list);
-        }
-
-        public static void Convert(ClassificationTypeMap typeMap, ITextSnapshot snapshot, List<ClassifiedSpan> list, Action<ITagSpan<IClassificationTag>> addTag)
-        {
-            foreach (var classifiedSpan in list)
-            {
-                addTag(Convert(typeMap, snapshot, classifiedSpan));
-            }
-        }
-
         public static TagSpan<IClassificationTag> Convert(ClassificationTypeMap typeMap, ITextSnapshot snapshot, ClassifiedSpan classifiedSpan)
         {
             return new TagSpan<IClassificationTag>(
                 classifiedSpan.TextSpan.ToSnapshotSpan(snapshot),
                 new ClassificationTag(typeMap.GetClassificationType(classifiedSpan.ClassificationType)));
         }
 
-        public static List<ITagSpan<IClassificationTag>> ConvertAndReturnList(ClassificationTypeMap typeMap, ITextSnapshot snapshot, List<ClassifiedSpan> classifiedSpans)
+        public static List<ITagSpan<IClassificationTag>> Convert(ClassificationTypeMap typeMap, ITextSnapshot snapshot, ArrayBuilder<ClassifiedSpan> classifiedSpans)
         {
-            var result = new List<ITagSpan<IClassificationTag>>();
-            Convert(typeMap, snapshot, classifiedSpans, result.Add);
-            ReturnClassifiedSpanList(classifiedSpans);
+            var result = new List<ITagSpan<IClassificationTag>>(capacity: classifiedSpans.Count);
```

Comment:
this value is passed back out to callers of ITagger.GetTags (which returns an enumerable.  In this case, it's fine to realize that as a List.

Reply:
I presume we're avoiding linq here for perf?